submissions/api.py

```diff
@@ -447,27 +447,27 @@ def get_all_course_submission_information(course_id, item_type, read_replica=Tru
     if read_replica:
         submission_qs = _use_read_replica(submission_qs)
 
-    query = submission_qs.select_related('student_item').prefetch_related('score_set').filter(
+    query = submission_qs.select_related('student_item__scoresummary__latest__submission').filter(
```

Comment:
As you stated, it really is a question of who needs the data download and what they want included:
1. Do they want all scores with additional data (like a latest flag) to help them filter the data in the download appropriately?  Or,
2. Does someone just want the latest (i.e. relevant) scores with no other historical scores?

One possibility is to go with the latest, since it is simpler, and add an option later if and when it is needed.  Can we find out if this aligns with the original requirement at a minimum?

@ormsbee @scottrish Thoughts on this?

Depending on which way we go, I'll have some additional minor requests on naming and commenting to make this more clear.